exercise/static/exercise/results_staff.js

```diff
@@ -704,163 +1004,496 @@
+                    _usertags[entry.name.toLowerCase()] = {color: entry.color, font_color: entry.font_color, name: entry.name, slug: entry.slug};
+                } else {
+                    _usertags[entry.id] = {color: entry.color, font_color: entry.font_color, name: entry.name, slug: entry.slug};
+                }
+                // Just save the bare minimum for rendering at this point
+            });
 
-        _students.forEach(function(student) {
-            const sid = student.id;
-            $.ajax(
-                $.extend({}, ajaxSettings, {url: pointsUrl + sid + '/'})
-            ).then(function(data) {
-                _points[sid] = data;
-                completedPointAjaxCalls++;
-                if (!successFirstStudent) {
-                    successFirstStudent = true;
-                    const firstStudentPoints = Object.values(_points)[0];
-                    firstStudentPoints.modules.forEach(function(module) {
-                        $("#module-selection").append(
-                            '<option value="module-' + module.id + '"'
+            // TODO: Get the link to students in a proper way
+            const participantsLink = $('li.menu-participants').find('a').attr('href');
+
+            // Dummy row is for static sorting (summary rows shown on top always)
+            let columns = [
+                /*{data: "dummy", title: "dummy", class: "always-hidden", searchable: true, 
+                    render: function(data, type, row, meta) {
+                        return (row['UserID'] === 0 ? Object.keys(summaries).length - meta.row : 0);
+                    } 
+                },*/
+                {data: "UserID", title: "UserID", class: "always-hidden col-0", type: "num", searchable: false},
+                {data: "Email", title: "Email", class: "always-hidden col-1", type: "string", searchable: true},
+                {data: "StudentID", title: "Student ID", type: "string", width: "6em", class: "student-id stick-on-scroll select-filter col-2" },
+                //{data: "FirstName", title: "FirstName"},
+                //{data: "LastName", title: "LastName"},
+                {data: "Name", title: "Student name", class: "student-name stick-on-scroll select-filter col-3", type: "html", render: function(data, type, row) { return (row['UserID'] > 0 ? '<a href="' + participantsLink + row['UserID'] + '">' + data + '</a>' : ''); } },
+                {data: "Tags", title: "Tags", class: "tags col-4", render: function(data) { return userTagsToHTML(data); }, type: "html" },
+                {data: "Count", title: "Count", class: "col-5", visible: false, defaultContent: 0, type: "num"},
+                {data: "Total", title: "Total", width: "6em", class: "total col-6", defaultContent: 0, type: "num"}
+            ];
+
+            // Store exercises globally
+            _exercises = exerciseJson[0].results;
```

Comment:
Can the default limit be overridden in the request? Seems like an easier solution than writing logic for multiple API calls.

Reply:
The pagination settings can be modified in the (Django REST framework) viewset class (in Python code). 
https://www.django-rest-framework.org/api-guide/pagination/

Did you test if there are actually any issues with the pagination? I haven't checked if the original CSV view has any pagination at all. It is possible to disable pagination. Pagination exists because some views would otherwise try to include far too much data in one response.